model/labels/regexp.go

```diff
@@ -63,13 +63,20 @@ func NewFastRegexMatcher(v string) (*FastRegexMatcher, error) {
 		// available, even if the string matcher is faster.
 		m.matchString = m.stringMatcher.Matches
 	} else {
-		parsed, err := syntax.Parse(v, syntax.Perl)
+		var parsed *syntax.Regexp
+		var err error
+		if v == ".*" || v == "^.*$" { // special case for .* to match everything
```

Comment:
Maybe I missed something in the conversation, but I am not expecting to see a special-case `^.*$` here.

(`.*` is also something of a special-case, but justified as the canonical regexp that someone would write when "anything" is meant)